src/server/services/hli/stadspas-gpass-service.ts

```diff
@@ -158,62 +158,57 @@ export async function fetchStadspassenByAdministratienummer(
 
   if (stadspasHouderResponse.status === 'ERROR') {
     return stadspasHouderResponse;
-  } else if (!stadspasHouderResponse.content) {
-    return NO_PASHOUDER_CONTENT_RESPONSE;
   }
 
   const pashouder = stadspasHouderResponse.content;
+  if (!pashouder) {
+    return NO_PASHOUDER_CONTENT_RESPONSE;
+  }
 
-  const pashouders = [
-    pashouder,
-    ...(stadspasHouderResponse.content?.sub_pashouders?.filter(Boolean) ?? []),
-  ].filter(
-    (p: StadspasHouderSource | null): p is StadspasHouderSource => p !== null
-  );
-
+  const pashouders = [pashouder, ...(pashouder.sub_pashouders ?? [])];
   const pasRequests = [];
 
   for (const pashouder of pashouders) {
-    // Filter out passes that are not relevant for the user.
-    const passen = pashouder.passen.filter((pas) => {
-      return (
-        pas.actief || (!pas.vervangen && isCurrentPasYear(pas.expiry_date))
-      );
-    });
-
-    for (const pas of passen) {
-      const response = fetchStadspasSource(
-        requestID,
-        pas.pasnummer,
-        administratienummer
-      ).then((response) => {
-        if (response.content && response.status === 'OK') {
-          const pasTransformed = transformStadspasResponse(
-            response.content,
-            pashouder,
-            pas.securitycode
-          );
-          const stadspas: ApiSuccessResponse<Stadspas> = {
-            ...response,
-            content: pasTransformed,
-          };
-
-          return stadspas;
-        }
-        return response;
-      });
-
-      pasRequests.push(response);
+    if (!pashouder) {
+      break;
+    }
```

Comment:
Mogelijk zijn de rest van de pashouders wel ok.. Je kunt dan beter `continue` gebruiken. Maar als je filtert op `Boolean` in mijn comment hierboven moet 't goed gaan zonder deze statement.